runbooks/CI_pipelines/adding-ci-dockerhub-users.md

```diff
@@ -74,11 +74,97 @@ there are 3 main steps:
 
 This user will have Member permissions, which will be enough to pull images. No higher permission level should be set.
 
-### 3. Add dockerhub credentials to CircleCI
-
-1. go to the project settings of the circleci build pipeline
-2. go to environment variables
-3. Add an environment variable for the service DockerID e.g. `$DOCKER_USER`
-4. Add an environment variable for the docker access token e.g. `$DOCKER_ACCESS_TOKEN`.
-5. You should now be able to reference these inside of the `config.yaml` for CircleCI.
-6. Follow the advice given in [using Docker Authenticated Pulls](https://circleci.com/docs/2.0/private-images/) to set up you pipeline with these credentials, when using environment variables.
+### 3. Add dockerhub credentials to a CircleCI Project
+
+1. Create AWS Secrets Manager secrets for the DockerID and access token in your product's module in opg-org-infra. Our intention is to create these secrets in the opg-management AWS account.
+
+    ```hcl
+    module "ci_dockerhub_secrets" {
+      source         = "../modules/ci_dockerhub_secrets"
+      product_prefix = local.product_prefix
+      providers      = { aws = aws.management }
+    }
+    output "aws_secretsmanager_secret_dockerhub_id" {
+      value = module.ci_dockerhub_secrets.aws_secretsmanager_secret_dockerhub_id
+    }
+    output "aws_secretsmanager_secret_dockerhub_token" {
+      value = module.ci_dockerhub_secrets.aws_secretsmanager_secret_dockerhub_token
+    }
+    ```
+
+2. Use the aws_secretsmanager_secret_version data source to provide the secret string as an environment variable to CircleCI. You can use the opg-org-infra repository module to create environment variables for CircleCI. Applying this module will set a value of `default` for each secret, to be replaced later.
+
+    ```hcl
+    data "aws_secretsmanager_secret_version" "product_dockerhub_id" {
+      secret_id = module.product.aws_secretsmanager_secret_dockerhub_id.id
+      provider  = aws.management
+    }
+
```

Comment:
Can you just add in the token here please as well :)

Reply:
done